### PR TITLE
refactored validators

### PR DESCRIPTION
**Pull request by dkudernatsch**

### CODE REVIEW

File: UnitTests/ExpectedException/ValidatorsTest.cs [hunk 1]
```diff
@@ -135,12 +144,9 @@ public void LastStatementIsExpressionStatementValidator_ValidatesWithoutError (s
...
     {
       var methodSyntax = CompiledSourceFileProvider.LoadMethod (c_validatorTestCasesFileName, method);
+      var validator = new ExpectedExceptionValidator();
 
-      var errors = Validators.ValidateMultiple (
-          methodSyntax,
-          new EmptyMethodBodyValidator(),
-          new AssertInLastStatementValidator(),
-          new LastStatementIsExpressionStatementValidator());
+      var errors = validator.Validate(methodSyntax).ToList();
```
Comment (dkudernatsch) [inline UnitTests/ExpectedException/ValidatorsTest.cs]:
```suggestion
      var errors = validator.Validate (methodSyntax).ToList();
```

File: UnitTests/ExpectedException/ValidatorsTest.cs [hunk 2]
```diff
@@ -151,12 +157,9 @@ public void AllAsserts_ValidateWithError (string method, params string[] reasons
...
     {
       var methodSyntax = CompiledSourceFileProvider.LoadMethod (c_validatorTestCasesFileName, method);
+      var validator = new ExpectedExceptionValidator();
 
-      var errors = Validators.ValidateMultiple (
-          methodSyntax,
-          new EmptyMethodBodyValidator(),
-          new AssertInLastStatementValidator(),
-          new LastStatementIsExpressionStatementValidator());
+      var errors = validator.Validate(methodSyntax);
```
Comment (dkudernatsch) [inline UnitTests/ExpectedException/ValidatorsTest.cs]:
```suggestion
      var errors = validator.Validate (methodSyntax);

```

File: UnitTests/ExpectedException/ValidatorsTest.cs [hunk 3]
```diff
@@ -34,9 +34,12 @@ public class ValidatorsTest
     public void MethodBodyValidator_ValidatesWithError (string method)
     {
       var methodSyntax = CompiledSourceFileProvider.LoadMethod (c_validatorTestCasesFileName, method);
-      var validator = new EmptyMethodBodyValidator();
+      var validator = new MethodBodyNotEmptyValidator();
 
-      var error = validator.Validate (methodSyntax);
+      var errors = validator.Validate (methodSyntax).ToList();
+
+      Assert.That(errors, Has.One.Items);
```
Comment (dkudernatsch) [inline UnitTests/ExpectedException/ValidatorsTest.cs]:
```suggestion
      Assert.That (errors, Has.One.Items);

```

File: Core/ExpectedException/ExpectedExceptionRewriter.cs [hunk 4]
```diff
@@ -41,15 +42,12 @@ public class ExpectedExceptionRewriter : CSharpSyntaxRewriter
...
     }
 
     public override SyntaxNode VisitMethodDeclaration (MethodDeclarationSyntax node)
     {
-      var errors = Validation.Validators.ValidateMultiple (
-          node,
-          new EmptyMethodBodyValidator(),
-          new AssertInLastStatementValidator(),
-          new LastStatementIsExpressionStatementValidator());
+      var errors = _validator.Validate(node).ToList();
```
Comment (dkudernatsch) [inline Core/ExpectedException/ExpectedExceptionRewriter.cs]:
```suggestion
      var errors = _validator.Validate (node).ToList();

```

File: UnitTests/ExpectedException/ValidatorsTest.cs [hunk 5]
```diff
@@ -151,14 +158,12 @@ public void AllAsserts_ValidateWithError (string method, params string[] reasons
...
-          new EmptyMethodBodyValidator(),
-          new AssertInLastStatementValidator(),
-          new LastStatementIsExpressionStatementValidator());
 
       Assert.That (errors, Is.Empty);
     }
   }
-}
\ No newline at end of file
+}
```
Comment (iam-mholle) [inline UnitTests/ExpectedException/ValidatorsTest.cs]: Unnecessary change

File: Core/ExpectedException/ExpectedExceptionRewriter.cs [hunk 6]
```diff
@@ -78,4 +77,4 @@ private IEnumerable<ExpectedExceptionModel> QueryExpectedExceptionAttributes (Ba
           .ToList();
     }
   }
-}
\ No newline at end of file
+}
```
Comment (iam-mholle) [inline Core/ExpectedException/ExpectedExceptionRewriter.cs]: Unnecessary change

**Review by iam-mholle — APPROVED** [review]
